### PR TITLE
bpo-10945: Drop support for bdist_wininst on non-Windows systems

### PR DESCRIPTION
**Pull request by hroncok**
bdist_wininst depends on MBCS codec, unavailable on non-Windows,
and bdist_wininst have not worked since at least Python 3.2, possibly
never on py3k.

Here we document that bdist_wininst is only supported on Windows
and we mark it _unsupported otherwise to skip tests.

Distributors of Python 3 can now safely drop the bdist_wininst .exe files
without the need to skip bdist_wininst related tests.

https://bugs.python.org/issue10945

### CODE REVIEW

File: Doc/distutils/builtdist.rst [hunk 1]
```diff
@@ -342,6 +341,9 @@ version number.  This can be changed to another text by using the
 The installer file will be written to the "distribution directory" --- normally
 :file:`dist/`, but customizable with the :option:`!--dist-dir` option.
 
+   .. versionchanged:: 3.9
```
Comment (vstinner) [inline Doc/distutils/builtdist.rst]: Please remove this note.

I don't see why this change should be Python 3.9 only: mbcs encoding was never available on platforms other than Windows. If we make a change, I suggest to make it in 3.7, 3.8 and master branches.

File: Lib/distutils/command/bdist_wininst.py [hunk 2]
```diff
@@ -55,6 +56,8 @@ class bdist_wininst(Command):
     boolean_options = ['keep-temp', 'no-target-compile', 'no-target-optimize',
                        'skip-build']
 
+    _unsupported = platform.system() != "Windows"
```
Comment (vstinner) [inline Lib/distutils/command/bdist_wininst.py]: I suggest:

```
     # bpo-10945: bdist_wininst requires mbcs encoding which is only available on Windows
     _unsupported = (sys.platform != "win32")
```

File: Misc/NEWS.d/next/Windows/2019-07-01-12-38-48.bpo-10945.s0YBHG.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Officially drop support for creating bdist_wininst installers on non-Windows
+systems.
```
Comment (vstinner) [inline Misc/NEWS.d/next/Windows/2019-07-01-12-38-48.bpo-10945.s0YBHG.rst:2]: Technically, it wasn't supported previously, so I don't think that we even have to document the change. But I'm fine with keeping this changelog entry if you prefer.

**Review by vstinner — CHANGES_REQUESTED** [review]

File: Doc/distutils/builtdist.rst (hunk 1 above)
Comment (hroncok) [reply Doc/distutils/builtdist.rst]: Gladly :rocket:

**Comment by hroncok** [comment]
I have made the requested changes; please review again

**Comment by hroncok** [comment]
An idea: Don't install the relevant exe files on non-Windows?

File: Lib/distutils/command/bdist_wininst.py [hunk 4]
```diff
@@ -4,6 +4,7 @@
 exe-program."""
 
 import sys, os
+import platform
```
Comment (vstinner) [inline Lib/distutils/command/bdist_wininst.py]: You can now removed this import which became unused.

Comment (hroncok) [reply Lib/distutils/command/bdist_wininst.py]: good catch

File: Doc/distutils/builtdist.rst [hunk 5]
```diff
@@ -315,8 +315,8 @@ or the :command:`bdist` command with the :option:`!--formats` option::
 
 If you have a pure module distribution (only containing pure Python modules and
 packages), the resulting installer will be version independent and have a name
-like :file:`foo-1.0.win32.exe`.  These installers can even be created on Unix
-platforms or Mac OS X.
+like :file:`foo-1.0.win32.exe`. Note that creating ``wininst`` binary
+distributions in only supported on Windows systems.
```
Comment (vstinner) [inline Doc/distutils/builtdist.rst:319]: That sounds better to me, thanks for the update!

File: Lib/distutils/command/bdist_wininst.py (hunk 4 above)
Comment (hroncok) [reply Lib/distutils/command/bdist_wininst.py]: done

**Comment by vstinner** [comment]
> An idea: Don't install the relevant exe files on non-Windows?

I would prefer to have a separated PR for that.

**Review by vstinner — APPROVED** [review]
LGTM. Thanks for the update.
